middleware/request_logger.go

```diff
@@ -83,13 +86,41 @@ func RequestLogger(logger *logrus.Logger) func(next http.Handler) http.Handler {
 					}.ServeHTTP(w, r)
 				}
 
+				status := ww.Status()
 				// Log the entry, the request is complete.
-				entry.Write(ww.Status(), ww.BytesWritten(), t2.Sub(t1))
+				createSubLog(r, status).
+					Int("status", status).
+					Int("size", ww.BytesWritten()).
+					Dur("duration", t2.Sub(t1)).
+					Msg("request complete")
 			}()
 
-			r = r.WithContext(lg.WithLogEntry(r.Context(), entry))
+			r = r.WithContext(entry.WithContext(r.Context()))
 			next.ServeHTTP(ww, r)
 		}
 		return http.HandlerFunc(fn)
 	}
 }
+
+func createSubLog(r *http.Request, status int) (subLog *zerolog.Event) {
+	logger := hlog.FromRequest(r)
+	if status >= 400 && status < 499 {
+		subLog = logger.Warn()
+	} else if status >= 500 {
+		subLog = logger.Error()
+	} else {
+		subLog = logger.Info()
+	}
+	for key, list := range r.Header {
```

Comment:
hmm let's drop the headers - we would currently be logging the auth token with this